### PR TITLE
Call static analysis finish endpoint after uploading files

### PR DESCRIPTION
**Pull request by joseph-sentry**
This commit changes the static-analysis command in the CLI, such that it calls the staticanalysis/analyses/<external_id>/finish endpoint in the API after it uploads the necessary files. This endpoint schedules the static analysis task on the worker immediately.

Fixes: #66

### CODE REVIEW

File: codecov_cli/services/staticanalysis/__init__.py [hunk 1]
```diff
@@ -139,6 +139,31 @@ async def run_analysis_entrypoint(
         )
     else:
         logger.info("All files are already uploaded!")
+    try:
```
Comment (dana-yaish) [inline codecov_cli/services/staticanalysis/__init__.py:142]: for readability reasons, can we extract this and put it in a function, maybe "send_finish_signal()"?

File: codecov_cli/services/staticanalysis/__init__.py [hunk 2]
```diff
@@ -139,6 +139,31 @@ async def run_analysis_entrypoint(
...
+        if response.status_code >= 400:
+            raise click.ClickException(
+                f"There is some problem with the submitted information.\n{response_json.get('detail')}"
+            )
+    except requests.RequestException:
+        raise click.ClickException(click.style("Unable to reach Codecov", fg="red"))
+    logger.info(
+        "Received response from server",
+        extra=dict(
+            extra_log_attributes=dict(time_taken=response.elapsed.total_seconds())
```
Comment (dana-yaish) [inline codecov_cli/services/staticanalysis/__init__.py:164]: I think it makes sense if we logged out the response status code and and the response text, wdyt?

Comment (joseph-sentry) [reply codecov_cli/services/staticanalysis/__init__.py:150]: makes sense!

File: codecov_cli/services/staticanalysis/__init__.py (hunk 1 above)
Comment (joseph-sentry) [reply codecov_cli/services/staticanalysis/__init__.py:142]: yes

**Review by dana-yaish — APPROVED** [review]
LGTM!

**Pull request merged**